### PR TITLE
Add support for generating metric data for testing

### PR DESCRIPTION
**Pull request by pmcollins**
This is in support of issue #652 and in response to a request by the team to break up my last commit into smaller chunks. This is the first of those chunks.

We will need to generate MetricData content both for feeding to the testbed sender and for testing MetricData diff functionality which will be used for validation checking in the testbed.

### CODE REVIEW

File: internal/goldendataset/metric_gen.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/data"
+)
+
+// Simple utilities for generating metrics for testing
+
+type MetricCfg struct {
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:26]: Please add a comment to all public structs/func/members :)

File: internal/goldendataset/metric_gen.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	MetricDescriptorType pdata.MetricType
+	NumBuckets           int
+	NumIlm               int
+	NumMetrics           int
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:30]: I think this is NumMetricsPerIl

File: internal/goldendataset/metric_gen.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/data"
+)
+
+// Simple utilities for generating metrics for testing
+
+type MetricCfg struct {
+	MetricDescriptorType pdata.MetricType
+	NumBuckets           int
+	NumIlm               int
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:29]: NumIlmPerResource?

File: internal/goldendataset/metric_gen.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+		bound := bounds[i]
+		if val <= bound {
+			bucket := buckets.At(i)
+			bucket.SetCount(bucket.Count() + 1)
+			break
+		}
+	}
+}
+
+func populateSummaryPoints(cfg MetricCfg, metric pdata.Metric) {
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:178]: May want to add quantiles/percentiles as well

File: internal/goldendataset/metric_gen.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+		v := cfg.PtVal + int64(i)
+		pt := pts.At(i)
+		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
+		pt.SetValue(float64(v))
+		populatePtLabels(cfg, pt.LabelsMap())
+	}
+}
+
+func populateHistogramPoints(cfg MetricCfg, metric pdata.Metric) {
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:142]: May want to add exemplars.

File: internal/goldendataset/metric_gen.go (hunk 1 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:26]: Done

File: internal/goldendataset/metric_gen.go (hunk 2 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:30]: Sorry, it turns out this is not used after a recent change. Removed.

File: internal/goldendataset/metric_gen.go (hunk 3 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:29]: Yes, fixed.

File: internal/goldendataset/metric_gen.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/goldendataset/metric_gen.go:26]: Probably you forgot to update the PR?

File: internal/goldendataset/metric_gen.go (hunk 4 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:189]: Done

File: internal/goldendataset/metric_gen.go (hunk 5 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:148]: Done

File: internal/goldendataset/metric_gen.go (hunk 1 above)
Comment (pmcollins) [reply internal/goldendataset/metric_gen.go:28]: Sorry, I should have waited to respond until the PR was ready. It's ready now and pushed.

File: internal/goldendataset/metric_gen.go [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+// MetricCfg holds parameters for generating dummy metrics for testing. Set values on this struct to generate
+// metrics with the corresponding number/type of attributes and pass into MetricDataFromCfg to generate metrics.
+type MetricCfg struct {
+	// The type of metric to generate
+	MetricDescriptorType pdata.MetricType
+	// The number of instrumentation library metrics per resource
+	NumIlmPerResource int
+	// The size of the MetricSlice and number of Metrics
+	NumMetrics int
```
Comment (bogdandrutu) [inline internal/goldendataset/metric_gen.go:34]: s/NumMetrics/NumMetricsPerIlm?

Comment (pmcollins) [reply internal/goldendataset/metric_gen.go]: Fixed : )

**Pull request merged**